vcd/writer.py

```diff
@@ -79,6 +79,7 @@ def __init__(self, file, timescale='1 us', date=None, comment='',
         self._scope_var_names = {}
         self._scope_types = {}
         self._ident_values = OrderedDict()
+        self._init_timestamp = init_timestamp
         self._prev_timestamp = 0
```

Comment:
The problematic line is this: 
```
        if timestamp and self._registering:
            self._finalize_registration()
```
We detect the end of registration by `timestamp` being non-zero.